src/modules/fundingManager/bondingCurve/abstracts/BondingCurveBase_v1.sol

```diff
@@ -238,8 +238,8 @@ abstract contract BondingCurveBase_v1 is IBondingCurveBase_v1, Module_v1 {
         // Cache Collateral Token
         IERC20 collateralToken = __Module_orchestrator.fundingManager().token();
 
-        // Transfer collateral, confirming that correct amount == allowance
-        collateralToken.safeTransferFrom(
+        // Handle collateral transfer
+        _handleCollateralTokensBeforeBuy(
             _msgSender(), address(this), _depositAmount
```

Comment:
the BondingCurveBase isnt the receiver of the buy function.
address (this) can just be used in the _handleCollateralTokensBeforeBuy function directly
We can either leave _receiver as part of the function call in (for potential added info, although we dont need it in this case or we can remove from the function altogether

Reply:
This doesn't seem correct to me. The `_receiver` references the receiver of the bought tokens (i.e. issuance tokens in this case), not the collateral tokens. With this change, the collateral tokens would never reach the contract, which they should! Or am I blind? @FHieser

Reply:
The idea was to adapt the _handleCollateralTokensBeforeBuy to use address(this) internally and take the receiver of the buy function as bonus information in case we need it somehow
But after talking to MarvinG about it we agreed to remove the receiver altogether